### ISSUE TITLE
Getting "404 page not found"

### ISSUE THREAD

**Issue by ljavierrivera** [wodby/docker4drupal#223]
This [search](https://github.com/wodby/docker4drupal/search?q=404+page+not+found&type=Issues&utf8=%E2%9C%93) did not turn up any resolutions.

I can't seem to get etc/hosts to load anything on browser either using 127.0.0.1 or docker-machine IP. However, when I run my docker-machine ip on port 8000 (x.x.x.x:8000), i get `404 page not found` in the Chrome.

### Codebase
Built-in vanilla Drupal
  
### Host OS 
macOS 10.12.5

### Docker info output
```yaml
Containers: 6
 Running: 6
 Paused: 0
 Stopped: 0
Images: 8
Server Version: 17.03.1-ce
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 75
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.4.57-boot2docker
Operating System: Boot2Docker 17.03.1-ce (TCL 7.2); HEAD : 4c264fa - Tue Mar 28 21:11:51 UTC 2017
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.858 GiB
Name: default
ID: Z64T:RU75:SUWG:JYJI:P7TE:I2UF:7LLS:XEZJ:YDBH:LSMD:NMSO:M7BV
Docker Root Dir: /mnt/sda1/var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 72
 Goroutines: 69
 System Time: 2017-12-15T22:14:20.700247587Z
 EventsListeners: 2
Username: ljavierrivera
Registry: https://index.docker.io/v1/
Labels:
 provider=virtualbox
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

### Docker compose file
```yml
Vanilla as per https://github.com/wodby/docker4drupal/commit/da8feae0d4a8b4a66e992bed3c5e021741681084
```

### Logs output
```bash
docker-compose logs mailhog
Attaching to docker4drupalfork_mailhog_1
mailhog_1    | 2017/12/15 22:14:04 Using in-memory storage
mailhog_1    | 2017/12/15 22:14:04 [SMTP] Binding to address: 0.0.0.0:1025
mailhog_1    | 2017/12/15 22:14:04 Serving under http://0.0.0.0:8025/
mailhog_1    | [HTTP] Binding to address: 0.0.0.0:8025
mailhog_1    | Creating API v1 with WebPath: 
mailhog_1    | Creating API v2 with WebPath: 
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
mailhog_1    | [APIv1] KEEPALIVE /api/v1/events
```

```bash
docker-compose logs mariadb
Attaching to docker4drupalfork_mariadb_1
mariadb_1    | Initializing database
mariadb_1    | 
mariadb_1    | PLEASE REMEMBER TO SET A PASSWORD FOR THE MariaDB root USER !
mariadb_1    | To do so, start the server, then issue the following commands:
mariadb_1    | 
mariadb_1    | '/usr/bin/mysqladmin' -u root password 'new-password'
mariadb_1    | '/usr/bin/mysqladmin' -u root -h  password 'new-password'
mariadb_1    | 
mariadb_1    | Alternatively you can run:
mariadb_1    | '/usr/bin/mysql_secure_installation'
mariadb_1    | 
mariadb_1    | which will also give you the option of removing the test
mariadb_1    | databases and anonymous user created by default.  This is
mariadb_1    | strongly recommended for production servers.
mariadb_1    | 
mariadb_1    | See the MariaDB Knowledgebase at http://mariadb.com/kb or the
mariadb_1    | MySQL manual for more instructions.
mariadb_1    | 
mariadb_1    | Please report any problems at http://mariadb.org/jira
mariadb_1    | 
mariadb_1    | The latest information about MariaDB is available at http://mariadb.org/.
mariadb_1    | You can find additional information about the MySQL part at:
mariadb_1    | http://dev.mysql.com
mariadb_1    | Consider joining MariaDB's strong and vibrant community:
mariadb_1    | https://mariadb.org/get-involved/
mariadb_1    | 
mariadb_1    | Database initialized
mariadb_1    | MySQL init process in progress...
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] mysqld (mysqld 10.2.11-MariaDB) starting as process 100 ...
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] InnoDB: Uses event mutexes
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] InnoDB: Compressed tables use zlib 1.2.11
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] InnoDB: Using Linux native AIO
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] InnoDB: Number of pools: 1
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] InnoDB: Using SSE2 crc32 instructions
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] InnoDB: Initializing buffer pool, total size = 1G, instances = 4, chunk size = 128M
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] InnoDB: Completed initialization of buffer pool
mariadb_1    | 2017-12-15 22:14:07 139853249817320 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] InnoDB: Highest supported file format is Barracuda.
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] InnoDB: 128 out of 128 rollback segments are active.
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] InnoDB: Creating shared tablespace for temporary tables
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] InnoDB: Waiting for purge to start
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] InnoDB: 5.7.20 started; log sequence number 1619994
mariadb_1    | 2017-12-15 22:14:07 139853277596392 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] Plugin 'FEEDBACK' is disabled.
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Warning] 'user' entry 'root@c2a410f94a6b' ignored in --skip-name-resolve mode.
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Warning] 'user' entry '@c2a410f94a6b' ignored in --skip-name-resolve mode.
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Warning] 'proxies_priv' entry '@% root@c2a410f94a6b' ignored in --skip-name-resolve mode.
mariadb_1    | 2017-12-15 22:14:07 139853277596392 [Note] InnoDB: Buffer pool(s) load completed at 171215 22:14:07
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] Reading of all Master_info entries succeded
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] Added new Master_info '' to hash table
mariadb_1    | 2017-12-15 22:14:07 139854723451720 [Note] mysqld: ready for connections.
mariadb_1    | Version: '10.2.11-MariaDB'  socket: '/var/run/mysqld/mysqld.sock'  port: 0  MariaDB Server
mariadb_1    | 2017-12-15 22:14:09 139853268376296 [Warning] 'proxies_priv' entry '@% root@c2a410f94a6b' ignored in --skip-name-resolve mode.
mariadb_1    | 2017-12-15 22:14:09 139853268376296 [Warning] 'proxies_priv' entry '@% root@c2a410f94a6b' ignored in --skip-name-resolve mode.
mariadb_1    | 
mariadb_1    | 2017-12-15 22:14:09 139853268056808 [Note] mysqld (unknown): Normal shutdown
mariadb_1    | 
mariadb_1    | 2017-12-15 22:14:09 139853268056808 [Note] Event Scheduler: Purging the queue. 0 events
mariadb_1    | 2017-12-15 22:14:09 139853278067432 [Note] InnoDB: FTS optimize thread exiting.
mariadb_1    | 2017-12-15 22:14:09 139853268056808 [Note] InnoDB: Starting shutdown...
mariadb_1    | 2017-12-15 22:14:09 139853277596392 [Note] InnoDB: Dumping buffer pool(s) to /var/lib/mysql/ib_buffer_pool
mariadb_1    | 2017-12-15 22:14:09 139853277596392 [Note] InnoDB: Buffer pool(s) dump completed at 171215 22:14:09
mariadb_1    | 2017-12-15 22:14:10 139853268056808 [Note] InnoDB: Shutdown completed; log sequence number 1620013
mariadb_1    | 2017-12-15 22:14:10 139853268056808 [Note] InnoDB: Removed temporary tablespace data file: "ibtmp1"
mariadb_1    | 2017-12-15 22:14:10 139853268056808 [Note] mysqld: Shutdown complete
mariadb_1    | 
mariadb_1    | 
mariadb_1    | MySQL init process done. Ready for start up.
mariadb_1    | 
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] mysqld (mysqld 10.2.11-MariaDB) starting as process 1 ...
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] InnoDB: Uses event mutexes
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] InnoDB: Compressed tables use zlib 1.2.11
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] InnoDB: Using Linux native AIO
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] InnoDB: Number of pools: 1
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] InnoDB: Using SSE2 crc32 instructions
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] InnoDB: Initializing buffer pool, total size = 1G, instances = 4, chunk size = 128M
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] InnoDB: Completed initialization of buffer pool
mariadb_1    | 2017-12-15 22:14:10 139997052672744 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] InnoDB: Highest supported file format is Barracuda.
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] InnoDB: 128 out of 128 rollback segments are active.
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] InnoDB: Creating shared tablespace for temporary tables
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] InnoDB: 5.7.20 started; log sequence number 1620013
mariadb_1    | 2017-12-15 22:14:10 139997081672424 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] Plugin 'FEEDBACK' is disabled.
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] Server socket created on IP: '0.0.0.0'.
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Warning] 'proxies_priv' entry '@% root@c2a410f94a6b' ignored in --skip-name-resolve mode.
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] Reading of all Master_info entries succeded
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] Added new Master_info '' to hash table
mariadb_1    | 2017-12-15 22:14:10 139998527527752 [Note] mysqld: ready for connections.
mariadb_1    | Version: '10.2.11-MariaDB'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  MariaDB Server
mariadb_1    | 2017-12-15 22:14:10 139997081672424 [Note] InnoDB: Buffer pool(s) load completed at 171215 22:14:10
```

```bash
docker-compose logs nginx
Attaching to docker4drupalfork_nginx_1
nginx_1      | 2017/12/15 22:14:05 [notice] 1#1: using the "epoll" event method
nginx_1      | 2017/12/15 22:14:05 [notice] 1#1: nginx/1.13.7
nginx_1      | 2017/12/15 22:14:05 [notice] 1#1: built by gcc 6.3.0 (Alpine 6.3.0) 
nginx_1      | 2017/12/15 22:14:05 [notice] 1#1: OS: Linux 4.4.57-boot2docker
nginx_1      | 2017/12/15 22:14:05 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
nginx_1      | 2017/12/15 22:14:05 [notice] 1#1: start worker processes
nginx_1      | 2017/12/15 22:14:05 [notice] 1#1: start worker process 24
nginx_1      | 2017/12/15 22:14:05 [notice] 1#1: start worker process 25
```

```bash
docker-compose logs portainer
Attaching to docker4drupalfork_portainer_1
portainer_1  | 2017/12/15 22:14:04 Starting Portainer 1.15.5 on :9000
```

```bash
docker-compose logs traefik
Attaching to docker4drupalfork_traefik_1
traefik_1    | time="2017-12-15T22:14:04Z" level=info msg="Using TOML configuration file /dev/null" 
traefik_1    | time="2017-12-15T22:14:04Z" level=info msg="Traefik version v1.4.5 built on 2017-12-06_10:16:48AM" 
traefik_1    | time="2017-12-15T22:14:04Z" level=info msg="Preparing server http &{Network: Address::80 TLS:<nil> Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc420504220} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s" 
traefik_1    | time="2017-12-15T22:14:04Z" level=info msg="Starting provider *docker.Provider {"Watch":true,"Filename":"","Constraints":null,"Trace":false,"DebugLogGeneratedTemplate":false,"Endpoint":"unix:///var/run/docker.sock","Domain":"","TLS":null,"ExposedByDefault":true,"UseBindPortIP":false,"SwarmMode":false}" 
traefik_1    | time="2017-12-15T22:14:04Z" level=info msg="Starting provider *web.Provider {"Address":":8080","CertFile":"","KeyFile":"","ReadOnly":false,"Statistics":null,"Metrics":null,"Path":"/","Auth":null,"Debug":false,"CurrentConfigurations":{},"Stats":{"Uptime":"2017-12-15T22:14:04.870247987Z","Pid":1,"ResponseCounts":{},"TotalResponseCounts":{},"TotalResponseTime":"0001-01-01T00:00:00Z"},"StatsRecorder":null}" 
traefik_1    | time="2017-12-15T22:14:04Z" level=info msg="Starting server on :80" 
traefik_1    | time="2017-12-15T22:14:04Z" level=info msg="Server configuration reloaded on :80" 
traefik_1    | time="2017-12-15T22:14:07Z" level=info msg="Server configuration reloaded on :80" 
```

**Comment by csandanov** [comment]
Are you using boot2docker? Try [docker for mac](https://www.docker.com/docker-mac)

**Comment by ljavierrivera** [comment]
Yes I'm using Docker Toolbox. I have other containers and don't want to run an upgrade just yet. any other options/suggestions?

**Comment by ljavierrivera** [comment]
After some tinkering, I got to a resolution on this repository / Docker Composition. It **WORKS**, however, documentation should be updated to address a few items:

1. **Do NOT use "extension" of "localhost** in the hosts file (/etc/hosts), instead use something like _drupal.docker.local_
    1. Chrome on Mac OS X doesn't propagate properly. Even though a ping to the domain resolves to the docker-machine's IP, chrome doesn't want to pick it up. however, changing the "extension" from localhost => local seemed to have "resolved" everything
2. **Update all instances of 'traefik.frontend.rule=Host:....' to match your new extension**
    1. For example: - 'traefik.frontend.rule=Host:portainer.drupal.docker.local'

when I updated the docker-compose to match items 1 & 2 above, everything was now working correctly.

Hope this helps someone else.
